src/odin/maneuversbuilder.cc

```diff
@@ -367,6 +368,37 @@ void ManeuversBuilder::Combine(std::list<Maneuver>& maneuvers) {
         curr_man = next_man;
         ++next_man;
       }
+      // Combine double L turns or double R turns in short non-internal intersections as u-turns
+      else if (!curr_man->internal_intersection() &&
+               curr_man->travel_mode() == TripLeg_TravelMode::TripLeg_TravelMode_kDrive &&
+               curr_man->length(Options::kilometers) <= (kMaxInternalLength * kKmPerMeter) &&
+               prev_man->HasSimilarNames(&(*next_man), true) && curr_man != next_man &&
+               !next_man->IsDestinationType()) {
+        if ((curr_man->type() == DirectionsLeg_Maneuver_Type_kLeft ||
```

Comment:
will need to check that the turn degree from the end of the prev man and the begin of the next man is reverse

Reply:
next_man->set_turn_degree(GetTurnDegree(prev_man->end_heading(), next_man->begin_heading()));